src/MUP_3/GeometricObjectException.java

```diff
@@ -1,15 +1,20 @@
 package MUP_3;
 
-import java.security.InvalidParameterException;
+import java.security.*;
 public class GeometricObjectException extends InvalidParameterException {
     //TODO warum einen leeren Konstruktor?
+    //das ist eine gute Frage :D
+    // in der Theorie kann man dadurch einfach nur schonmal die Klasse erstellen, ohne Message
+    // Problem hierbei ist, die msg / String den man der Elternklasse übergibt, kann man nur im Konstruktor setzen....
+    // -> ist nur da, weil der prof es will.
```

Comment:
Gibt ein prinzip, das nennt sich singelton /bean. Wird z.B. für Logger benutzt. Jede Klasse bekommt einen Logger in Konstruktor beim initzialisieren. Dafür wird meist ein leerer Konstruktor benutzt der Logger Klasse. 
Dadurch kann man im Code logs ausgeben und wenn man was für alle Klassen beim Logging geändert werden soll kann man einfach die Logger Klasse ändern.

ABER wie gesagt, hier irgendwie unverständlich der leere Konstruktor

Reply:
oder auch für Objekte die man genau einmal aufrufen muss, aber öfters braucht